python/tests/integration/memory/memory_stores/test_azure_cog_search_memory_store.py

```diff
@@ -114,8 +118,8 @@ async def test_record_not_found():
             assert False
         except MemoryConnectorResourceNotFound:
             pass
-
-        await memory_store.delete_collection(collection)
+        finally:
```

Comment:
Same here... we've already deleted the collection, and so we expect an exception and then we try and delete it again.